src/anomalib/data/folder.py

```diff
@@ -52,11 +53,24 @@ def make_folder_dataset(
     Returns:
         DataFrame: an output dataframe containing samples for the requested split (ie., train or test)
     """
-    normal_dir = _resolve_path(normal_dir, root)
-    abnormal_dir = _resolve_path(abnormal_dir, root) if abnormal_dir is not None else None
-    normal_test_dir = _resolve_path(normal_test_dir, root) if normal_test_dir is not None else None
-    mask_dir = _resolve_path(mask_dir, root) if mask_dir is not None else None
-    assert normal_dir.is_dir(), "A folder location must be provided in normal_dir."
+
+    def _path_to_list_config_with_resolve_path(path: str | Path | Sequence[str | Path] | None) -> List[Path]:
+        """Function for changing path to List[Path].
+        Args:
+            path (str | Path | Sequence | None): Path to replace with Sequence[str | Path].
+        Returns:
+            List[Path]: The result of path replaced by Sequence[str | Path].
+        """
```

Comment:
I always find this sort of stuff easier to follow when there is example section in the docstring. How about adding one like the following:

```suggestion
        """Convert path to list of paths.

        Args:
            path (str | Path | Sequence | None): Path to replace with Sequence[str | Path].

        Examples:
            >>> _path_to_list_config_with_resolve_path("dir")
            [Path("path/to/dir")]

            >>> _path_to_list_config_with_resolve_path(["dir1", "dir2"])
            [Path("path/to/dir1"), Path("path/to/dir2")]

        Returns:
            list[Path]: The result of path replaced by Sequence[str | Path].
        """
```